images/gpu-feature-discovery/tests/main.tf

```
@@ -0,0 +1,83 @@
+terraform {
+  required_providers {
+    oci       = { source = "chainguard-dev/oci" }
+    imagetest = { source = "chainguard-dev/imagetest" }
+  }
+}
+
+variable "digest" {
+  description = "The image digest to run tests over."
+}
+
+variable "name" {
+  default = "gpu-feature-discovery"
+}
+
+data "oci_string" "ref" { input = var.digest }
+
+data "imagetest_inventory" "this" {}
+
+resource "imagetest_harness_k3s" "this" {
+  name      = var.name
+  inventory = data.imagetest_inventory.this
+
+  sandbox = {
+    mounts = [
+      {
+        source      = path.module
+        destination = "/tests"
+      }
+    ]
+  }
+}
+
+module "helm" {
+  source    = "../../../tflib/imagetest/helm"
+  name      = "nvgfd"
+  repo      = "https://nvidia.github.io/gpu-feature-discovery"
+  chart     = "gpu-feature-discovery"
+  namespace = "gpu-feature-discovery"
+
+  values = {
+    image = {
+      repository = data.oci_string.ref.registry_repo
+      tag        = "unused@${element(split("@", var.digest), 1)}"
```

```suggestion
      tag        = data.oci_string.ref.pseudo_tag
```

This should work as well, it comes from the same input digest variable and is a bit cleaner

---

Definitely use this, I've got it working locally with this tag, and removing the helm commands block.